dlib/simd/simd8f_vec.h

```diff
@@ -0,0 +1,251 @@
+// Copyright (C) 2013  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_sIMD8F_Hh_
```

Comment:
I would say keep the separate files, except for these that are copies of other files.  As for using .cpp files, that would be fine if all compilers performed link time optimization as well as GCC but I doubt that's the case.  And I don't want to require the use of CMake so anything that makes compilation easier for users is better.

On that note, if you want to split the current simd8 code into a C and SSE .h files that's fine with me.  Or leaving them alone and just #including the simd8 as needed is fine too.  My main concern is to avoid unnecessary code duplication.